.github/workflows/delete-untagged-images.yml

```diff
@@ -0,0 +1,41 @@
+name: Delete untagged images in ECR
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "58 1 * * *" # runs everyday at 1:58AM
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read # This is required for actions/checkout
+
+jobs:
+  cleanup:
+    strategy:
+      matrix:
+        environment: ["dev"] # , "staging", "prod"
+        repository: ["graasp", "graasp/explore"]
+    name: Cleanup ${{ matrix.repository }} on ${{ matrix.environment }}
+    runs-on: ubuntu-latest
+    environment: ${{ matrix.environment }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE }}
+          role-session-name: manageECRactionJob
+          aws-region: ${{ vars.AWS_REGION }}
+
+      - name: Get images
+        run: echo "IMAGE_DIGESTS=$(aws ecr list-images --repository-name ${{ matrix.repository }} --filter tagStatus=UNTAGGED | jq -r '.imageIds | map("imageDigest=\(.imageDigest)") | join(" ")')" >> $GITHUB_ENV
+
+      - name: Delete images
+        if: ${{ env.IMAGE_DIGESTS != '' }}
+        run: |
+          result=$(aws ecr batch-delete-image --repository-name ${{ matrix.repository }} --image-ids ${{ env.IMAGE_DIGESTS }})
+          echo $result
+          echo $result | jq -r '"Deleted \(.imageIds | length) images, failed to delete \(.failures | length) images."' >> $GITHUB_STEP_SUMMARY
```

Comment:
I would add more docs about what images are fetched

Reply:
I have added some comments above each step. Let me know if you would like more documentation and in which form.